### ISSUE TITLE
in-cluster operator fails to install control plane after having deleted a prior control plane

### ISSUE THREAD

**Issue by dgn** [istio/istio#21467]
**Bug description**

Basically we don't seem to cleanup the cache on removal.

```
2020-02-25T17:02:32.131059Z	info	Generated manifest objects are the same as cached for component Galley.
2020-02-25T17:02:32.334041Z	info	Generated manifest objects are the same as cached for component IngressGateways.
2020-02-25T17:02:32.528521Z	info	Generated manifest objects are the same as cached for component EgressGateways.
2020-02-25T17:02:33.629894Z	info	Generated manifest objects are the same as cached for component Pilot.
2020-02-25T17:02:37.140414Z	info	Generated manifest objects are the same as cached for component AddonComponents.
```

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
1.5.0-beta.3

It works if you kill the pod and let it restart

**Comment by morvencao** [comment]
Is this resolved by https://github.com/istio/istio/pull/21020 ?

**Comment by sdake** [comment]
I have tested this works as expected. Closing. Nice job @morvencao !

**Comment by sdake** [comment]
To be clear to folks that find this issue via a google search, This is fixed in the upcoming 1.5.1 release of Istio.
